src/azul/indexer.py

```diff
@@ -34,12 +34,16 @@ class BaseIndexer(ABC):
     def mapping(self) -> JSON:
         raise NotImplementedError()
 
-    def settings(self) -> JSON:
+    def settings(self, index_name) -> JSON:
+        # Setting a large number of shards for the contributions indexes (i.e. not aggregate) greatly speeds up indexing
+        # which is our biggest bottleneck, however doing the same for the aggregate index dramatically limits searching.
+        # This was because every search had to check every shard and nodes became overburdened with so many requests.
+        # Instead we try using one shard per instance which is optimal for searching since it allows parallelization of
+        # requests (though maybe at the cost of some speed during indexing).
+        num_shards = config.es_instance_count if '_aggregate' in index_name else config.indexer_concurrency
```

Comment:
Use `config.parse_es_index_name`. You always want to keep the implementation of composing and resource decomposing names in one place. In the long run, that makes changing the naming scheme much easier. Also, `'_aggregate' in` is a little weak. The aformentioned method is stricter.

Reply:
Yes it is. I've changed to that.